support/k8s/k8s-workload-registrar/mode-crd/controllers/identity_schema.go

```diff
@@ -0,0 +1,313 @@
+	// }
+	// if testCase.configAnnotation != "" && testCase.podAnnotation != "" {
+	// 	pod.Annotations[testCase.configAnnotation] = testCase.podAnnotation
+	// }
+
+	// Test:
+	//spiffeID := c.podSpiffeID(pod)
+
+	finalId := is.getSVID(pod)
+	log.Printf("** Final id %v", finalId)
+	log.Printf("Identity %#v", is)
+	fmt.Print(&is)
+}
+
+func (is *IdentitySchema) getSVID(pod *corev1.Pod) string {
+
+	log.Printf("identity_schema, getId: processing Pod %s", pod.Name)
+
+	var idString string = ""
+	fields := is.Fields
+	for i, field := range fields {
+		log.Printf("identity_schema, getId: %d Field name: %s", i, field.Name)
+		idString += "/" + is.getFieldValue(pod, field)
+	}
+	log.Printf("identity_schema, getSVID: ID Value: %s", idString)
+	return idString
+}
+
+func (is *IdentitySchema) getFieldValue(pod *corev1.Pod, field Field) string {
+
+	att := field.AttestorSource
+	if att != nil {
+		log.Printf("* identity_schema, getFieldValue: Field Attestor Group Name: %v", att.Group)
+		value, err := att.GetValue(pod)
+		if err != nil {
+			log.Printf("* identity_schema, getFieldValue: Error processing the field %s, with attestor source %v", field.Name, err)
+			// TODO for now, when error return the field name
+			return field.Name
+		}
+		return value
+	}
+
+	cm := field.ConfigMapSource
+	if cm != nil {
+		log.Printf("* identity_schema, getFieldValue: ConfigMap Name %s", cm.Name)
+		log.Printf("* identity_schema, getFieldValue: ConfigMap Field %s", cm.Field)
+		log.Printf("* identity_schema, getFieldValue: ConfigMap Namespace %s", cm.Namespace)
+		value, err := cm.GetValue(pod)
+		if err != nil {
+			log.Printf("* identity_schema, getFieldValue: Error processing the field %s, with configMap source: %v", field.Name, err)
+			// TODO for now, when error return the field name
+			return field.Name
+		}
+		return value
+	}
+
+	// TODO for now if value unknown, just return the field name
+	log.Printf("* identity_schema, getFieldValue: Error processing the field %s, no matching source!", field.Name)
+	return field.Name
+}
+
+func (att *AttestorSource) GetValue(pod *corev1.Pod) (value string, err error) {
+
+	log.Printf("** identity_schema, GetValue(Attestor): pod: %s", pod.Name)
+	// log.Printf("** This attestor uses mapping: %#v", attestor.Mapping)
+
+	switch att.Group {
+	case nodeAttestor:
+		log.Print("**identity_schema, GetValue: Processing nodeAttestor")
+		return "value-from-node-Attestor", nil
+	case workloadAttestor:
+		// here we don't need the selector values. We just need the value for the SPIFFE ID
+		_, value, err = att.getValueFromWorkloadAttestor(pod)
+		if err != nil {
+			log.Printf("%s", err)
+			return value, err
+		} else {
+			return value, nil
+		}
+	default:
+		log.Print("** identity_schema, GetValue: Unknown attestor name")
+		err := fmt.Errorf("Unknown attestor name: %s", att.Group)
+		return value, err
+	}
+}
+
+func (cms *ConfigMapSource) GetValue(pod *corev1.Pod) (value string, err error) {
```

Comment:
we may need to revamp this interface a bit to make it easy to pass the client, either through some context or passing in a client specifically